src/coreclr/jit/codegenriscv64.cpp

```diff
@@ -6227,17 +6227,19 @@ void CodeGen::genJmpPlaceVarArgs()
 //
 void CodeGen::genIntCastOverflowCheck(GenTreeCast* cast, const GenIntCastDesc& desc, regNumber reg)
 {
+    const regNumber tempReg = internalRegisters.GetSingle(cast);
+
     switch (desc.CheckKind())
     {
         case GenIntCastDesc::CHECK_POSITIVE:
         {
-            genJumpToThrowHlpBlk_la(SCK_OVERFLOW, INS_blt, reg, nullptr, REG_R0);
+            GetEmitter()->emitIns_R_R(INS_sext_w, EA_8BYTE, tempReg, reg);
```

Comment:
```suggestion
            GetEmitter()->emitIns_R_R(INS_sext_w, EA_4BYTE, tempReg, reg);
```
Nit. The parameter is unused, but *w instructions are EA_4BYTE everywhere.